### ISSUE TITLE
#| comments |# font-lock failure on emacs snapshot

### ISSUE THREAD

**Issue by greghendershott** [greghendershott/racket-mode#220]
As of [build 500](https://travis-ci.org/greghendershott/racket-mode/builds/134870402) for commit d991d10 the font-lock test for `#| multi-line comments |#` started [failing](https://travis-ci.org/greghendershott/racket-mode/jobs/134870417#L2532-L2533) _only_ on emacs head.

However commit d991d10 doesn't change anything related to font-lock. And, multi-line font-lock still works fine on e.g. emacs 24.3 and 24.5.

Did something change on emacs head to break our syntax table (or be more strict about our syntax table which was somehow wrong all along)?

```
    ;; Comment related
    (modify-syntax-entry ?\; "<   " st) ;line comments but NOT sexp #;
    (modify-syntax-entry ?\n ">   " st)

    (modify-syntax-entry ?#  "w 14" st) ;not necessarily prefix
    (modify-syntax-entry ?|  "_ 23bn" st)
```

**Comment by greghendershott** [comment]
Although this test still fails on `snapshot`, it passes on the recently-released 25.1: [example build results](https://travis-ci.org/greghendershott/racket-mode/builds/162225649).

It also passes when I `make test-elisp` locally with 25.1 on macOS. Furthermore the `#| comments #|` font-lock correctly when manually inspecting `racket-mode` buffers.

As a result:

1. I'm not sure what's going on. Is emacs `snapshot` configured differently than release builds, in a way that impacts this corner case of font-lock??

2. Although I don't like the test failure and I want to understand what's happening, this _seems_ to be N/A for real users: This _seems_ to be about `snapshot` and not about 25?

**Comment by greghendershott** [comment]
At some point `snapshot` stopped doing this so I'm going to close this as non-actionable.
